modules/appeals_api/app/controllers/concerns/appeals_api/pdf_downloads.rb

```diff
@@ -0,0 +1,142 @@
+# frozen_string_literal: true
+
+require 'common/file_helpers'
+require 'pdf_fill/filler'
+
+# rubocop:disable Metrics/ModuleLength
+module AppealsApi
+  module PdfDownloads
+    extend ActiveSupport::Concern
+
+    included do
+      def render_appeal_pdf_download(appeal, filename)
+        if expired?(appeal)
+          # If the veteran_icn is available, we can check it against the current request's ICN header to return a 404
+          # if it does not match, but otherwise, without auth_headers, we can't tell whether the current request is
+          # authorized, so in that case we return a specific error message regardless of authorization status.
+          if appeal.veteran_icn && appeal.veteran_icn != request.headers['X-VA-ICN']
+            raise ActiveRecord::RecordNotFound
+          else
+            return render_pdf_download_expired(appeal)
```

Comment:
Since we save the `veteran_icn` on the appeal record via the `AddIcnUpdater` job, it may be possible to correctly 404 when the ICN on an expunged record doesn't match the request header, but if we don't have that and we don't have `auth_headers`, I don't know of way to determine whether to return a 404 or a 422 here. Here I'm returning a 422 with an 'expired' message even if they're not authorized in that specific case, but I was thinking that might be ok, since the error message at least doesn't expose PII and they have already supplied the UUID of the appeal - What do others think?

Reply:
I wish I was more familiar with the `AddIcnUpdater` job to give a suggestion. When the data is expunged, the `veteran_icn` still sticks around, right? Which records/situations wouldn't have the `veteran_icn` populated?

I'm not confident that `422` is the right error code for the `else` clause, based on the [Lighthouse standards error code flowchart](https://department-of-veterans-affairs.github.io/lighthouse-api-standards/errors/#choosing-an-error-code). It seems like a `404` would be the correct code based on that chart, but that also seems less than ideal unless it's okay to return the `404` and also explain to the consumer the reason why it wasn't found (expiry) alongside it.